src/cmdline.cpp

```diff
@@ -0,0 +1,79 @@
+// local headers
+#include "cmdline.h"
+#include "utility.h"
+
+CmdlineArgs::CmdlineArgs() :
+    m_parser{"Tool to check and set file permissions"},
+    system_mode{"", "system", "system mode, act according to /etc/sysconfig/security", m_parser},
+    force_fscaps{"", "fscaps", "force use of file system capabilities", m_parser},
+    disable_fscaps{"", "no-fscaps", "disable use of file system capabilities", m_parser},
+    apply_changes{"s", "set", "actually apply changes (--system mode may imply this depending on file based configuration)", m_parser},
+    only_warn{"", "warn", "only inform about which changes would be performed but don't actually apply them (which is the default, except in --system mode)", m_parser},
+    no_header{"n", "noheader", "don't print intro message", m_parser},
+    verbose{"v", "verbose", "print additional output that might be useful for diagnosis", m_parser},
+    print_variables{"", "print-variables", "print supported profile variable mappings then exit", m_parser},
+    examine_paths{"e", "examine", "operate only on the specified path(s)", false, "PATH", m_parser},
+    force_level_list{"", "level", "force application of the specified space-separated list of security levels (only supported in --system mode)", false, "", "e.g. \"local paranoid\"", m_parser},
+    file_lists{"f", "files", "read newline separated list of files to check (see --examine) from the specified path", false, "PATH", m_parser},
+    root_path{"r", "root", "check files relative to the given root directory", false, "", "PATH", m_parser},
+    config_root_path{"", "config-root", "lookup configuration files relative to the given root directory", false, "", "PATH", m_parser},
+    input_args{"args", "in --system mode a list of paths to check, otherwise a list of profiles to parse", false, "PATH", m_parser} {
+
+    }
+
+int CmdlineArgs::parse(const int argc, const char **argv) {
+    m_parser.parse(argc, argv);
+
+    return validateArguments() ? 0 : 2;
```

Comment:
Process exit codes are rather arbitrary unless they are specifically documented. Except that `0` is generally interpreted as successful termination of the process.

What is attempted here, I guess, is differentiating between a usage error (i.e. wrong command line arguments supplied) and an operational error of any other kind. To make this clearer constant can be introduced. If you want that then I would add this as another commit on top of the PR. Ideally this would also be documented in the man page than. But I doubt that anybody would be evaluating that. It could rather be a debugging aid if `chkstat` fails in whatever circumstances and we can only see the exit code in the logs, for example.